File: 538/convert-bst-to-greater-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var convertBST = function (root) {
    let total = 0;
    function dfs(node) {
        if (!node) return;
        else {
            dfs(node.right);
            node.val += total;
            total = node.val;
            dfs(node.left);
        }
    }
    dfs(root);
    return root;
};

File: 538/convert-bst-to-greater-tree.py
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None


class Solution:
    def convertBST(self, root):
        """
        :type root: TreeNode
        :rtype: TreeNode
        """
        def dfs(node):
            if node is None:
                return
            else:
                dfs(node.right)
                node.val += dfs.total
                dfs.total = node.val
                dfs(node.left)

        dfs.total = 0
        dfs(root)
        return root
        